engine/src/main/java/org/terasology/persistence/internal/EntitySetDeltaRecorder.java

```diff
@@ -83,9 +92,12 @@ private EntityDelta getOrCreateEntityDeltaFor(EntityRef entity) {
         return entityDelta;
     }
 
-    public void onEntityDestroyed(long entityId) {
-        entityDeltas.remove(entityId);
-        destroyedEntities.add(entityId);
+    public void onEntityDestroyed(EntityRef entity) {
+        if (!entity.isPersistent()) {
```

Comment:
Not an issue, but could be rewritten as
```java
if (entity.isPersistent()) {
   // do stuff
}
```
to avoid multiple points of return from the method (and save a line of code).